src/control/lib/spdk/src/nvme_control_common.c

```diff
@@ -248,28 +248,48 @@ _discover(prober probe, bool detach, health_getter get_health)
 	return ret;
 }
 
+/* Assign ASCII string as defined by the NVMe spec */
 static int
-copy_ctrlr_data(struct ctrlr_t *cdst, const struct spdk_nvme_ctrlr_data *cdata)
+copy_ascii(char *dst, const void *buf, size_t buf_sz)
 {
-	int	written;
+	const uint8_t	*str = buf;
+	int		 i = 0;
 
-	written = snprintf(cdst->model, sizeof(cdst->model), "%-20.20s",
-			   cdata->mn);
-	if (written >= sizeof(cdst->model))
+	if (buf_sz >= BUFLEN)
```

Comment:
This strikes me as a bit odd. I'd expect equality to BUFLEN to be okay. I noticed  in the caller that we're getting the size of the incoming buffers with sizeof, which is right to me. But in that case, even if the dest size == BUFLEN, src size == BUFLEN could be fine, as the actual string we come up with could be shorter than the buffer length.

I had to read on to realize you were using buf_sz to mean the length of the string inside the buf. One suggestion, for clarity in differentiating the two: add a new variable called len, that starts at buf_sz - 1, and use len to represent the string length.

Reply:
addressed